source/cursorManager.py

```diff
@@ -365,14 +368,23 @@ def script_selectAll(self,gesture):
 		self._selectionMovementScriptHelper(toPosition=textInfos.POSITION_ALL)
 
 	def script_copyToClipboard(self,gesture):
-		info=self.makeTextInfo(textInfos.POSITION_SELECTION)
-		if info.isCollapsed:
-			# Translators: Reported when there is no text selected (for copying).
-			ui.message(_("No selection"))
+		info = self.makeTextInfo(textInfos.POSITION_SELECTION)
+		if not info.isCollapsed:
+			if info.copyToClipboard():
+				# Translators: Message presented when text has been copied to clipboard.
+				ui.message(_("Copied to clipboard"))
 			return
-		if info.copyToClipboard():
-			# Translators: Message presented when text has been copied to clipboard.
-			ui.message(_("Copied to clipboard"))
+		if isinstance(self, TreeInterceptor):
+			info = self.rootNVDAObject.makeTextInfo(textInfos.POSITION_SELECTION)
+			if not info.isCollapsed:
+				# There is indeed a selection, but L{info} strangely enough
+				# does not contains it (tested with Firefox and Chrome).
+				KeyboardInputGesture.fromName("control+c").send()

```

Comment:
Any reason why not using this instead?

```suggestion
				gesture.send()
```

Reply:
It was in case another gesture had been bound to this script, but I realize the script is not advertised with a description, and thus this is unlikely to happen.